pymeasure/instruments/ametek/ametek7270.py

```diff
@@ -0,0 +1,203 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2017 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import logging
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import modular_range, truncated_discrete_set, truncated_range
+
+class Ametek7270(Instrument):

```

Comment:
Looks good!

Reply:
All of the code as written should work using a VISA GPIB adapter. I'm currently using this machine over ethernet, so I first tried using VISA since it should be able to connect to things over TCP/IP. I ran into trouble with this, and I think it is because of the strange termination characters that the machine returns (for the IP interface, the 7270 returns three termination characters and under normal operation two of them will be ASCII 0 characters, which I could not get to work in VISA. You can set custom termination characters, but VISA gave me an error which seemed to say that there could not be two of the same character in a terminaton sequence). To get around this, I wrote a custom adapter using the Python built-in `socket` class, but since it seems to be pretty specific I'm not sure if you would want it included in "official PyMeasure."

Reply:
Interesting. Your socket adapter could be a good addition in the future. I will keep it in mind if another user raises an issue on the subject.

Reply:
yeah, changing used/expected terminators (for all adapters) would be a great addition for pymeasure, tbh. I 'm currently fighting with two instruments that insist on terminators that seem to be non-default, so it's difficult to change them without replacing pymeasure internals.